### ISSUE TITLE
[Conditional AR] Initialize auxiliary embeddings at Llama scale

### ISSUE THREAD

**Issue by gkondas** [mmcdermott/EveryQuery-conditional#16]
## Summary

`ConditionalQueryARModel` constructs three auxiliary embedding tables after constructing the Hugging Face `LlamaModel`:

```python
self.HF_model = LlamaModel._from_config(self.HF_model_config, **extra_kwargs)

self.answer_embed = torch.nn.Embedding(N_ANSWER_CLASSES, H)
self.token_type_embed = torch.nn.Embedding(N_TOKEN_TYPES, H)
self.block_pos_embed = torch.nn.Embedding(max_queries, H)
```

The Llama parameters are initialized using `config.initializer_range` (default `0.02`), but these later, outer-module `nn.Embedding` instances retain PyTorch's default initialization, `normal_(weight)` (approximately `N(0, 1)`).

This creates a roughly 50x standard-deviation mismatch between the Llama code embeddings and each added auxiliary embedding.

Relevant code: https://github.com/mmcdermott/EveryQuery-conditional/blob/claude/decoder-only-llamma-model-lm1vvk/src/every_query/model/conditional_ar_model.py

## Why this matters

The model adds these embeddings directly:

```python
patient_emb = code_emb + token_type_emb
query_code_emb = code_emb + token_type_emb + block_pos_emb
```

At initialization, the shared token-type/block-position vectors can therefore dominate the actual clinical-code embedding. In variance terms, a component with std `1.0` contributes about 2,500x the variance of one with std `0.02`.

This may suppress code identity in the initial normalized representations and make optimization especially difficult for deeper configurations. Empirically, on the same training data:

- tiny AR (2 layers, hidden 64) reduces tuning loss from about 0.133 to 0.086;
- large AR (12 layers, hidden 768) remains near 0.132–0.133 through roughly 10k optimizer steps.

The initialization mismatch is a plausible explanation for that size-dependent behavior, but the training observation alone does not establish causality.

## Documentation/source support

- Hugging Face documents `LlamaConfig.initializer_range=0.02` as the standard deviation used to initialize weight matrices: https://huggingface.co/docs/transformers/model_doc/llama#transformers.LlamaConfig
- `LlamaModel.__init__` calls `self.post_init()`, so that initialization only visits modules already inside the Llama model: https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py
- PyTorch `nn.Embedding.reset_parameters()` calls `init.normal_(self.weight)`, whose defaults are mean 0 and std 1: https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/sparse.py

## Proposed fix

Explicitly initialize the three auxiliary tables using the Llama initializer scale after creating them:

```python
std = self.HF_model_config.initializer_range
for embedding in (
    self.answer_embed,
    self.token_type_embed,
    self.block_pos_embed,
):
    torch.nn.init.normal_(embedding.weight, mean=0.0, std=std)
```

Do not call `apply` or `post_init` over the whole wrapper after this point, since that could reinitialize the already-created Llama backbone.

## Verification

1. Add a unit test checking that all three auxiliary embedding tables have sample standard deviations close to `config.initializer_range`.
2. Log the initial standard deviations of the Llama input embeddings and auxiliary embeddings in a smoke test.
3. Repeat the tiny/large comparison after the fix, ideally with the same gradient accumulation/effective batch size so the optimizer-step budgets are matched.
